userguide/tutorials/plugin_installation.adoc

```diff
@@ -0,0 +1,259 @@
+"http://127.0.0.1:8080/1.0/kb/nodesInfo"
+----
+
+Note that this also requires starting the plugin explicitly using the following command:
+
+[source,bash]
+----
+curl -v \
+-u admin:password \
+-H "Content-Type: application/json" \
+-H 'X-Killbill-CreatedBy: stephane' \
+-X POST \
+--data-binary '{"isSystemCommandType":true,"nodeCommandType":"RESTART_PLUGIN","nodeCommandProperties":[{"key":"pluginKey","value":"payment-test"}]}' \
+"http://127.0.0.1:8080/1.0/kb/nodesInfo"
+----
+
+=== Deploying by Hand
+
+Deploying by hand consists of building the self contained OSGI jar, and copying that jar at the right location. For example, the `adyen` plugin with a version with version `0.3.2` would show up as the following:
+
+```
+java
+|_adyen-plugin
+  |_ 0.3.2
+    |_ adyen-plugin-0.3.2.jar
+```
+
+[[plugin_configuration]]
+== Plugin Configuration
+
+=== System Properties
+
+Kill Bill plugins can access Kill Bill properties. Both system properties passed to the JVM and properties from the https://docs.killbill.io/latest/userguide_configuration.html#global_configuration_properties[_Kill Bill configuration file_] are accessible to the plugins and can be used to tweak the behavior of the plugin as needed.
+
+=== Configuration File
+
+A plugin can also specify its own property files which can be used to configure global settings for a plugin. Such property files need to be part of the archive (the OSGI mechanism will make sure these are only visible to the particular plugin):
+
+* The property file needs to be on the classpath (`resource` directory)
+* There is no restriction on the format of the property file, but typically  plugins will rely on `key-value` properties, `json` or `xml` files.
+
+[[per-tenant-configuration]]
+=== Per-tenant Configuration
+
+The two previous mechanisms work well for global settings, but are inadequate to configure the plugins on a per-tenant fashion (e.g for a payment plugin interacting with a payment gateway, different credentials might be needed for different tenants). In those situations, Kill Bill provides APIs to upload/retrieve/delete per-tenant plugin configurations.
+
+At a high level, this mechanism works in the following way:
+
+1. An administrator uses the Kill Bill API (or Kaui) to upload the configuration.
+1. Kill Bill stores the config in the `tenant_kvs` table using a `tenant_key` of `PLUGIN_CONFIG_<pluginName>` and sets the `tenant_value` with the config provided.
+1. Kill Bill broadcasts the change across the cluster of nodes and emits a configuration bus event: `TENANT_CONFIG_CHANGE` or `TENANT_CONFIG_DELETION`.
+4. The plugin code is *responsible to listen to these events* and take appropriate action to reload/delete its configuration for that specific tenant.
```

Comment:
We might want to highlight/explain how it's done in the hello world plugin for instance: https://github.com/killbill/killbill-hello-world-java-plugin/blob/3aa938d19fdfba81c7c035b45c3f17cac74db177/src/main/java/org/killbill/billing/plugin/helloworld/HelloWorldActivator.java#L56

Reply:
I will make a note of this, I'm anyway planning to add code snippets corresponding to all the different [plugin configuration methods](https://docs.killbill.io/latest/plugin_installation.html#plugin_configuration), so will add it at that time.

Reply:
Sounds good, thanks!